### PR TITLE
Remove getContentPointer usage from search 

### PR DESCRIPTION
**Pull request by taranvohra**
This PR removes usage of `getContentPointer` in the Search components. They are being passed through props. I also took this opportunity to cleanup some legacy code such as support for Collections etc. More to come in later PRs

### CODE REVIEW

File: packages/gitbook/src/app/(space)/(content)/[[...pathname]]/page.tsx [hunk 1]
```diff
@@ -94,7 +94,7 @@ export default async function Page(props: {
                 {page.layout.outline ? (
                     <PageAside
                         space={space}
-                        site={parent?.object === 'site' ? parent : undefined}
+                        site={site?.object === 'site' ? site : undefined}
```
Comment (SamyPesse) [inline packages/gitbook/src/app/(space)/(content)/[[...pathname]]/page.tsx:97]: Why not

```suggestion
                        site={site}
```

File: packages/gitbook/src/components/Header/CompactHeader.tsx [hunk 2]
```diff
@@ -20,15 +20,12 @@ import { SearchButton } from '../Search';
  */
 export function CompactHeader(props: {
     space: Space;
-    parent: Site | Collection | null;
+    site: Site | null;
```
Comment (SamyPesse) [inline packages/gitbook/src/components/Header/CompactHeader.tsx:23]: How is it possible to have a header without a `site` ?

File: packages/gitbook/src/lib/seo.ts [hunk 3]
```diff
@@ -55,16 +49,12 @@ export function isSpaceIndexable({
...
+        return shouldIndexVisibility(site.visibility);
     }
 
-    if (space.visibility === ContentVisibility.InCollection) {
-        return parent && parent.object === 'collection'
-            ? shouldIndexVisibility(parent.visibility)
-            : false;
-    }
-    return shouldIndexVisibility(space.visibility);
+    // space with no site should not be indexed
```
Comment (SamyPesse) [inline packages/gitbook/src/lib/seo.ts:56]: What is a space with no site?

Or is it stuff that you want to change in another PR for the pdf export?

**Review by SamyPesse — APPROVED** [review]

File: packages/gitbook/src/lib/seo.ts (hunk 3 above)
Comment (taranvohra) [reply packages/gitbook/src/lib/seo.ts:56]: yes

File: packages/gitbook/src/components/Header/CompactHeader.tsx (hunk 2 above)
Comment (taranvohra) [reply packages/gitbook/src/components/Header/CompactHeader.tsx:23]: `Header` is rendered by `SpaceLayout` which can currently render without a site (ex - for  pdf export )

**Pull request merged**